camayoc/tests/qcs/cli/test_sources.py

```diff
@@ -0,0 +1,879 @@
+    )
+    qpc_source_edit.logfile = BytesIO()
+    assert qpc_source_edit.expect(
+        'Source "{}" does not exist.'.format(invalid_name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 1
+
+
+def test_edit_port(isolated_filesystem, qpc_server_config):
+    """Edit a source's port.
+
+    :id: dc218224-7618-4bec-abc6-39c904290c11
+    :description: Edit the port of a source entry.
+    :steps: Run ``qpc source edit --name <name> --port <newport>``
+    :expectedresults: The source's port must be updated.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    hosts = '127.0.0.1'
+    port = new_port = random.randint(0, 65535)
+    while port == new_port:
+        new_port = random.randint(0, 65535)
+    cred_add(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --port {} '
+        '--type network'
+        .format(name, cred_name, hosts, port)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    source_show(
+        {'name': name},
+        '{{\r\n'
+        '    "credentials": \[\r\n'
+        '        {{\r\n'
+        '            "id": \\d+,\r\n'
+        '            "name": "{}"\r\n'
+        '        }}\r\n'
+        '    \],\r\n'
+        '    "hosts": \[\r\n'
+        '        "{}"\r\n'
+        '    \],\r\n'
+        '    "id": \\d+,\r\n'
+        '    "name": "{}",\r\n'
+        '    "port": {},\r\n'
+        '    "source_type": "network"\r\n'
+        '}}\r\n'
+        .format(cred_name, hosts, name, port)
+    )
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --port {}'
+        .format(name, new_port)
+    )
+    assert qpc_source_edit.expect(
+        'Source "{}" was updated'.format(name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 0
+
+    source_show(
+        {'name': name},
+        '{{\r\n'
+        '    "credentials": \[\r\n'
+        '        {{\r\n'
+        '            "id": \\d+,\r\n'
+        '            "name": "{}"\r\n'
+        '        }}\r\n'
+        '    \],\r\n'
+        '    "hosts": \[\r\n'
+        '        "{}"\r\n'
+        '    \],\r\n'
+        '    "id": \\d+,\r\n'
+        '    "name": "{}",\r\n'
+        '    "port": {},\r\n'
+        '    "source_type": "network"\r\n'
+        '}}\r\n'
+        .format(cred_name, hosts, name, new_port)
+    )
+
+
+def test_edit_port_negative(isolated_filesystem, qpc_server_config):
+    """Edit the port of a not created source entry.
+
```

Comment:
Suggestion: "Edit port of a source entry that does not exist"

Reply:
Much better, thank you